File: leetcode/twosum.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        std::vector<int> result;

        return result;
    }

};


int pow(int base, int index) {
    int result = 1;
    
    for (int i = 0; i < index; i++) {
        result *= base;
    }

    return result;
}


int main (int argc, char *argv[]) {
    Solution sol;
    int target;

    std::cout << "enter a target" << std::endl;
    std::cin >> target;

    if (target > pow(10, 9) || target < pow(10, 9) * -1) {
        std::cout << "target out of range" << std::endl;
        return 1;
    }

    return 0;
}

File: leetcode/twosum.java
class Solution {
    static int indexof(int[] arr, int target) {
        for (int i = 0; i < arr.length; i++) {
            if (target == arr[i]) {
                return i;
            }
        }

        return -1;
    }

    public int[] twoSum(int[] nums, int target) {
        int result[] = new int[2];

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            int complementindex = indexof(nums, complement);

            if (complementindex != -1 && complementindex != i) {
                result[0] = i;
                result[1] = complementindex;

                return result;
            }
        }

        return new int[] { -1, -1 };
    }
}


